apps/ci/ci-install.sh

```diff
@@ -2,59 +2,82 @@
 
 set -e
 
-cat >>conf/config.sh <<CONFIG_SH
-MTHREADS=$(expr $(grep -c ^processor /proc/cpuinfo) + 2)
-CWARNINGS=ON
-CDEBUG=OFF
-CTYPE=Release
-CSCRIPTS=ON
-CUNIT_TESTS=ON
-CSERVERS=ON
-CTOOLS=ON
-CSCRIPTPCH=OFF
-CCOREPCH=OFF
-CCUSTOMOPTIONS='-DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_FLAGS="-Werror" -DCMAKE_CXX_FLAGS="-Werror"'
-DB_CHARACTERS_CONF="MYSQL_USER='root'; MYSQL_PASS='root'; MYSQL_HOST='localhost';"
-DB_AUTH_CONF="MYSQL_USER='root'; MYSQL_PASS='root'; MYSQL_HOST='localhost';"
-DB_WORLD_CONF="MYSQL_USER='root'; MYSQL_PASS='root'; MYSQL_HOST='localhost';"
-CONFIG_SH
-
-time sudo apt-get update -y
-# time sudo apt-get upgrade -y
-time sudo apt-get install -y git lsb-release sudo ccache
-time ./acore.sh install-deps
+COMPILATION_OPTIONS='-DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache'
+# These flags below are disabled for gcc, gcc 10 and clang 10.
+ADDITIONAL_OPTIONS='-DCMAKE_C_FLAGS="-Werror" -DCMAKE_CXX_FLAGS="-Werror"'
+C_COMP=Unknown
+CPP_COMP=Unknown
 
 case $COMPILER in
+  # this is in order to use the "default" gcc version of the OS, without forcing a specific version
+  "gcc" )
+    time sudo apt-get install -y gcc g++
+    C_COMP=gcc
+    CPP_COMP=g++
+    ;;
+
+  "gcc10" )
+    time sudo apt-get install -y gcc-10 g++-10
+    C_COMP=gcc-10
+    CPP_COMP=g++-10
+    ;;
 
   # this is in order to use the "default" clang version of the OS, without forcing a specific version
   "clang" )
     time sudo apt-get install -y clang
-    echo "CCOMPILERC=\"clang\"" >> ./conf/config.sh
-    echo "CCOMPILERCXX=\"clang++\"" >> ./conf/config.sh
+    C_COMP=clang
+    CPP_COMP=clang++
+    COMPILATION_OPTIONS="${COMPILATION_OPTIONS} ${ADDITIONAL_OPTIONS}"
     ;;
 
   "clang6" )
     time sudo apt-get install -y clang-6.0
-    echo "CCOMPILERC=\"clang-6.0\"" >> ./conf/config.sh
-    echo "CCOMPILERCXX=\"clang++-6.0\"" >> ./conf/config.sh
+    C_COMP=clang-6.0
+    CPP_COMP=clang++-6.0
+    COMPILATION_OPTIONS="${COMPILATION_OPTIONS} ${ADDITIONAL_OPTIONS}"
     ;;
 
   "clang9" )
     time sudo apt-get install -y clang-9
-    echo "CCOMPILERC=\"clang-9\"" >> ./conf/config.sh
-    echo "CCOMPILERCXX=\"clang++-9\"" >> ./conf/config.sh
+    C_COMP=clang-9
+    CPP_COMP=clang++-9
+    COMPILATION_OPTIONS="${COMPILATION_OPTIONS} ${ADDITIONAL_OPTIONS}"
     ;;
 
   "clang10" )
     time sudo apt-get install -y clang-10
-    echo "CCOMPILERC=\"clang-10\"" >> ./conf/config.sh
-    echo "CCOMPILERCXX=\"clang++-10\"" >> ./conf/config.sh
-    # disable -Werror for clang-10 TODO: remove when this is fixed https://github.com/azerothcore/azerothcore-wotlk/issues/3108
```

Comment:
I'd preserve this TODO somewhere, and possibly open an issue for GCC as well and linking it here. So we don't forget :)